engine/graphics/src/vulkan/graphics_vulkan.cpp

```diff
@@ -453,14 +454,17 @@ namespace dmGraphics
             vk_usage_flags,
             vk_memory_type,
             vk_aspect_flags,
-            VK_IMAGE_LAYOUT_UNDEFINED,
             depth_stencil_texture_out);
         CHECK_VK_ERROR(res);
 
         if (res == VK_SUCCESS)
         {
-            res = TransitionImageLayout(vk_device, context->m_LogicalDevice.m_CommandPool, context->m_LogicalDevice.m_GraphicsQueue, depth_stencil_texture_out->m_Handle.m_Image, vk_aspect_flags,
-                VK_IMAGE_LAYOUT_UNDEFINED, VK_IMAGE_LAYOUT_DEPTH_STENCIL_ATTACHMENT_OPTIMAL);
+            res = TransitionImageLayout(vk_device,
+                context->m_LogicalDevice.m_CommandPool,
+                context->m_LogicalDevice.m_GraphicsQueue,
+                depth_stencil_texture_out,
+                vk_aspect_flags,
+                VK_IMAGE_LAYOUT_DEPTH_STENCIL_ATTACHMENT_OPTIMAL);
```

Comment:
As an FYI, textures should already know the current layout now, so there's no need to pass in the "current layout" anymore. This is so that we can put barriers when needed before binding a texture to a shader during the CommitUniforms phase.